lib/FeedItem.php

```diff
@@ -428,6 +428,37 @@ public function setUid($uid) {
 		return $this;
 	}
 
+	/**
+	 * Get duration (for items where the link points to media)
+	 *
+	 * Use {@see FeedItem::setDuration()} to set the duration.
+	 *
+	 * @return int Duration in seconds.
+	 */
+	public function getDuration() {
+		return $this->duration;
+	}
+
+	/**
+	 * Set duration (for items where the link points to media)
+	 *
+	 * Use {@see FeedItem::getDuration()} to get the duration.
+	 *
+	 * @param int $duration Duration of the media in seconds
+	 * @return self
+	 */
+	public function setDuration($duration) {
+		$this->duration = null;
+
+		if(!is_int($duration) || $duration < 0) {
+			Debug::log('Duration must be a non-negative integer!');
```

Comment:
I would go as far as raising an [`InvalidArgumentException`](https://www.php.net/manual/en/class.invalidargumentexception.php).

Reply:
Though that would require using non-legacy items in the bridges that would want to set the duration:

Something like:

```php
$item = new FeedItem([
    'title' => ...,
    ...,
]);

try {
    $item->setDuration(
} catch (InvalidArgumentException $e)) {
    Debug::log($e->getMessage();
}
```

Otherwise adding an item with invalid duration would crash.

Reply:
I would rather not complicate the setting of this variable in bridges too much.